report/map_helpers.go

```diff
@@ -148,19 +148,21 @@ func (m StringLatestMap) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
 func (m StringLatestMap) Less(i, j int) bool { return m[i].key < m[j].key }
 
 // sort entries and shuffle down any duplicates
-func (m StringLatestMap) fixup() {
-	sort.Sort(m)
-	for i := 1; i < len(m); {
-		if m[i-1].key == m[i].key {
-			if m[i-1].Timestamp.Before(m[i].Timestamp) {
-				m = append(m[:i-1], m[i:]...)
+func (m *StringLatestMap) fixup() {
```

Comment:
hmm. maybe make it return a slice instead. Would simplify the call sites too: `return out.fixup()`.

Downside is that it might give the appearance of being a pure function.

Reply:
See if a different verb case helps.